demandlib/bdew.py

```diff
@@ -115,21 +115,40 @@ def create_bdew_load_profiles(self, dt_index, slp_types, holidays=None):
         new_df.drop('date', axis=1, inplace=True)
         return new_df.div(new_df.sum(axis=0), axis=1)
 
-    def get_profile(self, ann_el_demand_per_sector):
+    def get_profile(self, ann_el_demand_per_sector,
+                    dyn_function_h0: bool = True):
         """ Get the profiles for the given annual demand
 
         Parameters
         ----------
         ann_el_demand_per_sector : dictionary
             Key: sector, value: annual value
+        dyn_function_h0: bool, default True
+            Uses the dynamisation function of the BDEW to smoothen the
+            seasonal edges. Functions resolution is daily.
+            f(x) = -3.916649251 * 10^-10 * x^4 + 3,2 * 10^-7 * x³ - 7,02
+            * 10^-5 * x²+0,0021 * x +1,24
+            Adjustment of accuracy: from -3,92 to -3.916649251
 
         Returns
         -------
         pandas.DataFrame : Table with all profiles
 
         """
-        return self.slp_frame.multiply(pd.Series(
-            ann_el_demand_per_sector), axis=1).dropna(how='all', axis=1) * 4
+        if dyn_function_h0 == True:
+            quartersinyear = len(self.slp_frame)
+            for quarter in range(quartersinyear):
```

Comment:
This way of doing it is rather inefficient. You should rather calculate an array of all smoothing factors and multiply the SLP by that. The following might work (did not verify):
```Python
day_of_year = [(quarter + 1) / (24 * 4) for quarter in range(len(self.slp_frame))]
smoothing_factor = [-3.916649251 * 10 ** -10
                                   * timestep ** 4 + 3.2 * 10 ** -7
                                   * timestep ** 3 - 7.02 * 10 ** -5
                                   * timestep ** 2 + 0.0021
                                   * timestep + 1.24 for timestep in day_of_year]
self.slp_frame['h0'][quarter] = self.slp_frame['h0'] * smoothing_factor
```
(Or - preferably - using numpy.)